### PR TITLE
Add dotnet

### PR DESCRIPTION
**Pull request by vreynolds**

### CODE REVIEW

**Review by paulosman — APPROVED** [review]
I could even run it! :tada:

File: Tiltfile [hunk 1]
```diff
@@ -67,22 +67,52 @@ def launch_java_year_service(auto_init=True):
...
+def launch_dotnet_name_service(auto_init=True):
+    launch_dotnet_svc(".NET-name", dirname="dotnet/name-service", auto_init=auto_init)
+
+def launch_dotnet_year_service(auto_init=True):
+    launch_dotnet_svc(".NET-year", dirname="dotnet/year-service", auto_init=auto_init)
 
 # Launch one of each of these types of services. Go services init by default
 # launch_go_frontend()
 # launch_python_frontend()
-launch_ruby_frontend()
```
Comment (MikeGoldsmith) [inline Tiltfile:74]: Should we have lines for each lang? This replaces ruby with .NET.

File: Tiltfile [hunk 2]
```diff
@@ -67,22 +67,52 @@ def launch_java_year_service(auto_init=True):
...
 
 # launch_go_name_service() # doesn't recognize Java autoinstr
 launch_python_name_service()
 # launch_ruby_name_service()
+# launch_dotnet_name_service()
 
 # launch_go_year_service()
 # launch_python_year_service()
-launch_java_year_service()
+# launch_dotnet_year_service()
```
Comment (MikeGoldsmith) [inline Tiltfile:115]: Same as above, should we remove java here?

File: dotnet/frontend/Startup.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+using Microsoft.OpenApi.Models;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+namespace frontend
+{
+    public class Startup
+    {
+        private const string ActivitySourceName = "honeycomb.examples.frontend-dotnet";
+        public static readonly ActivitySource ActivitySource = new(ActivitySourceName);
```
Comment (MikeGoldsmith) [inline dotnet/frontend/Startup.cs:18]: Wow - that must be a new lang shorthand, it's strange!

**Review by MikeGoldsmith — APPROVED** [review]
Looks good, just a couple of minor queries.

File: Tiltfile (hunk 1 above)
Comment (robbkidd) [reply Tiltfile:74]: Curses! My edit went weird. I only meant to add, not remove.

Comment (robbkidd) [reply Tiltfile:74]: Switched things back to ALL THE LANGUAGES and Go as the default (despite this being a .NET branch).

File: dotnet/frontend/Properties/launchSettings.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      "launchBrowser": true,
+      "launchUrl": "swagger",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "frontend": {
+      "commandName": "Project",
+      "dotnetRunMessages": "true",
+      "applicationUrl": "http://localhost:5003",
```
Comment (robbkidd) [inline dotnet/frontend/Properties/launchSettings.json:23]:
```suggestion
      "applicationUrl": "http://localhost:7000",
```

File: dotnet/message-service/Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      "launchBrowser": true,
+      "launchUrl": "swagger",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "message_service": {
+      "commandName": "Project",
+      "dotnetRunMessages": "true",
+      "applicationUrl": "http://localhost:5002",
```
Comment (robbkidd) [inline dotnet/message-service/Properties/launchSettings.json:23]:
```suggestion
      "applicationUrl": "http://localhost:9000",
```

File: dotnet/name-service/Properties/launchSettings.json [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      "launchBrowser": true,
+      "launchUrl": "swagger",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "name_service": {
+      "commandName": "Project",
+      "dotnetRunMessages": "true",
+      "applicationUrl": "http://localhost:5001",
```
Comment (robbkidd) [inline dotnet/name-service/Properties/launchSettings.json:23]:
```suggestion
      "applicationUrl": "http://localhost:8000",
```

File: dotnet/year-service/Properties/launchSettings.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+      "launchUrl": "swagger",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "year_service": {
+      "commandName": "Project",
+      "dotnetRunMessages": "true",
+      "launchUrl": "api/year",
+      "applicationUrl": "http://localhost:5000",
```
Comment (robbkidd) [inline dotnet/year-service/Properties/launchSettings.json:24]:
```suggestion
      "applicationUrl": "http://localhost:6001",
```

File: dotnet/frontend/Controllers/GreetingController.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+            var httpClient = _clientFactory.CreateClient();
+            var name = await GetNameAsync(httpClient);
+            var message = await GetMessage(httpClient);
+
+            return $"Hello {name}, {message}";
+        }
+
+        private static async Task<string> GetNameAsync(HttpClient client)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, "http://localhost:5001/name");
```
Comment (robbkidd) [inline dotnet/frontend/Controllers/GreetingController.cs:35]:
```suggestion
            var request = new HttpRequestMessage(HttpMethod.Get, "http://localhost:8000/name");
```

File: dotnet/frontend/Controllers/GreetingController.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+            var response = await client.SendAsync(request);
+            if (!response.IsSuccessStatusCode) return "OH NO!";
+            return await response.Content.ReadAsStringAsync();
+        }
+
+        private static async Task<string> GetMessage(HttpClient client)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, "http://localhost:5002/message");
```
Comment (robbkidd) [inline dotnet/frontend/Controllers/GreetingController.cs:43]:
```suggestion
            var request = new HttpRequestMessage(HttpMethod.Get, "http://localhost:9000/message");
```

File: dotnet/name-service/Controllers/NameController.cs [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+
+        [HttpGet]
+        public async Task<string> GetAsync()
+        {
+            var current = Activity.Current;
+            current?.AddTag("apple", 1);
+            current?.AddBaggage("avocado", "12");
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "http://localhost:5000/year");
```
Comment (robbkidd) [inline dotnet/name-service/Controllers/NameController.cs:39]:
```suggestion
            var request = new HttpRequestMessage(HttpMethod.Get, "http://localhost:6001/year");
```

**Review by robbkidd — COMMENTED** [review]
To have the .NET apps listen on and use the same ports as the equivalent service in the other language implementations.

**Pull request merged**